woocommerce/woocommerce-android
# Feature/6618 limit stats refresh

⚠️ Do not merge until [FluxC changes are merged ](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2526)

Part of: #6618 


### Description
`wc-analytics` endpoint is one of the slowest endpoints we hit from the app, and is one that we hit a lot. We fetch site stats and top performers every time we open My Store tab and when we change between tabs. This translates into the "slow/endless loading" perception of the app. This PR alongside changes applied in this other PR in woocommerce-android aims to reduce that loading "slowness" for **top performer products data** and provide better offline support  with the following changes: 

- Migrate top performers cache to Room to easily expose an observable
- Simplify top performers cached model to avoid having to parse multiple JSON or query different sql tables to load top performers from cache.
- Invalidate top performers cache when a new order is placed 
- Invalidate top performers cache after 1h. Maybe this is to often? please share your thoughts

Finally, I updated all broken unit tests, remove some that did not apply anymore, and added a couple more. 

### Testing instructions
Log into the app and check the following: 
- Top performers are loaded correctly for all stats granularities 
- Top performers are invalidated when a new order notification is received. Check for logcat message`Invalidating top performer products` after receiving the notification
- Check that after data is invalidated when changing stat granularity, the data is fetched from API and refeshed accordingly with new values.. See the log `fetchLeaderboards` is displayed in logcat as well as notice how the top performer list is refershed
- Check that once all 4 tabs data is loaded at least once, switching between tabs doesn't trigger fetching data from API. Basically, check that `fetchLeaderboards` is never called
- Check that after swipe to refresh is called, then switching between tabs will refresh data fro API
- Check that when we try to refresh top performer data without connectivity, when connectivity is back (enable/disable airplane mode for example) the top performer list is refreshed with data from API.
